sklearn/svm/_base.py

```diff
@@ -1140,8 +1140,8 @@ def _fit_liblinear(
     intercept_ : float
         The intercept term added to the vector.
 
-    n_iter_ : int
-        Maximum number of iterations run across all classes.
+    n_iter_ : array of int
+        Number of iterations run across for each class.
```

Comment:
Is it only impacting LR and LRCV ?

Reply:
`LinearSVR` and `LinearSVC` needs to be adjusted here:

https://github.com/scikit-learn/scikit-learn/blob/ec3bb7f4b33c358ca75ff0ab20564c6c58eea5d9/sklearn/svm/_classes.py#L257

and take the max, to be consistent with their docstring:

https://github.com/scikit-learn/scikit-learn/blob/ec3bb7f4b33c358ca75ff0ab20564c6c58eea5d9/sklearn/svm/_classes.py#L126-L127

Reply:
I agree but I would rather not change `LinearSVC` and `LinearSVR` because they only use the liblinear solver and their docstrings only documents the `n_iter_` attribute as a scalar so changing the shape of the `n_iter_` attribute might not really qualify as a bugfix anymore.

For logistic regression, the shape was always documented to be the one implemented in this PR, without mention of the solver.

Reply:
But changing the type of what ``fit_liblinear`` returns doesn't require any modification in the linearSVC/R code ?

Reply:
In this PR, the `n_iter_` attribute changes to a ndarray:

```python
from sklearn.svm import LinearSVC
from sklearn.datasets import make_classification

X, y = make_classification(random_state=42, n_classes=3, n_informative=4)
svc = LinearSVC().fit(X, y)
svc.n_iter_
# array([1000, 1000,  856], dtype=int32)
```

I think we can do a update:

https://github.com/scikit-learn/scikit-learn/blob/ec3bb7f4b33c358ca75ff0ab20564c6c58eea5d9/sklearn/svm/_classes.py#L257

To:

```python
self.coef_, self.intercept_, n_iter_ = _fit_liblinear(...)

self.n_iter_ = np.max(n_iter_)
```

to be backward compatible.

Reply:
Indeed. I had not read that code properly. We also need a new test it seems.

Reply:
IIUC, can this conversation be marked as resolved?